# Segunda parte : Analisis

## Paso 1: Esta celda importa Pandas y carga el dataset

In [1]:
import pandas as pd
df = pd.read_csv('dataset_limpio.csv')

# Verificar las dimensiones y las primeras filas
print(f"El dataset contiene {df.shape[0]} filas y {df.shape[1]} columnas.")
display(df.head(3))

El dataset contiene 17647 filas y 23 columnas.


,año,no_doc,universidad,tipo,especialidad/subespecialidad,modalidad,serum,sncds,1er_niv,5to_sup,...,not_final,ingresó_si/no,universidad_(ingreso),tipo_(ingreso),especialidad/subespecialidad_(ingreso),modalidad_(ingreso),sede_(ingreso),total_bonificacion,rango_pregrado,nota_cv
0,2023,73815684,CONAREME,Esp.,ANATOMIA PATOLOGICA,Libre,10.0,NaN,0.0,0.0,...,73.870,SI,UNMSM,Esp.,ANATOMIA PATOLOGICA,Libre,Hospital Nacional Edgardo Rebagliati Martins,10.0,Bajo,13.470
1,2023,71701411,CONAREME,Esp.,ANATOMIA PATOLOGICA,Libre,10.0,NaN,0.0,0.0,...,71.173,SI,UNMSM,Esp.,ANATOMIA PATOLOGICA,Libre,Hospital Nacional Edgardo Rebagliati Martins,10.0,Bajo,13.973
2,2023,46024669,CONAREME,Esp.,ANATOMIA PATOLOGICA,Libre,10.0,NaN,0.0,1.0,...,69.893,SI,URP,Esp.,ANATOMIA PATOLOGICA,Libre,Hospital Nacional Edgardo Rebagliati Martins,11.0,Bajo,13.893


## Paso 2: Generación de Resúmenes Estadísticos Matemáticos

### calculamos las medidas de tendencia central y dispersión pura para las variables críticas.

In [2]:
# 1. Definir las variables numéricas de interés
columnas_numericas = ['examen', 'enam', 'not_final', 'nota_cv', 'total_bonificacion']

print("--- 1. Medidas de Tendencia Central y Dispersión (General) ---")
# Generamos el resumen estadístico general (media, mediana, desviación estándar, rangos)
resumen_estadistico = df[columnas_numericas].describe().round(2)
display(resumen_estadistico)

print("\n--- 2. Resumen Estadístico Agrupado por Condición de Ingreso ---")
# Comparamos matemáticamente los promedios de los que ingresaron vs los que no
resumen_ingreso = df.groupby('ingresó_si/no')[columnas_numericas].mean().round(2)
display(resumen_ingreso)

# 3. Calculamos la Varianza matemática de las notas finales
print("\n--- 3. Varianza de la Nota Final ---")
varianza_notas = df['not_final'].var()
print(f"La varianza poblacional de la Nota Final es: {varianza_notas:.2f}")

--- 1. Medidas de Tendencia Central y Dispersión (General) ---


,examen,enam,not_final,nota_cv,total_bonificacion
count,17647.00,17647.00,17647.00,17647.00,17647.00
mean,49.46,1.30,61.71,12.25,9.16
std,7.74,0.34,8.73,2.05,1.88
min,10.00,1.00,23.08,1.00,0.00
25%,44.00,1.00,55.39,10.88,8.00
50%,49.60,1.00,61.82,12.76,10.00
75%,55.20,1.50,68.14,13.46,10.00
max,77.60,2.00,91.10,18.98,15.00



--- 2. Resumen Estadístico Agrupado por Condición de Ingreso ---


,examen,enam,not_final,nota_cv,total_bonificacion
ingresó_si/no,,,,,
NO,47.60,1.26,59.65,12.05,9.02
SI,52.23,1.37,64.79,12.56,9.36



--- 3. Varianza de la Nota Final ---
La varianza poblacional de la Nota Final es: 76.24


## Paso 3: Análisis de Correlaciones

###  Generamos la matriz matemática para validar qué variable tiene mayor peso sobre el ingreso.

In [3]:
print("--- Matriz de Correlación de Pearson ---")
# Calculamos la correlación entre las variables numéricas
matriz_correlacion = df[columnas_numericas].corr().round(3)
display(matriz_correlacion)

# Validamos específicamente la correlación con la Nota Final
print("\n--- Correlación directa con la Nota Final ---")
correlacion_notafinal = matriz_correlacion['not_final'].sort_values(ascending=False)
display(correlacion_notafinal)

--- Matriz de Correlación de Pearson ---


,examen,enam,not_final,nota_cv,total_bonificacion
examen,1.000,0.531,0.976,0.381,0.284
enam,0.531,1.000,0.569,0.416,0.237
not_final,0.976,0.569,1.000,0.573,0.480
nota_cv,0.381,0.416,0.573,1.000,0.974
total_bonificacion,0.284,0.237,0.480,0.974,1.000



--- Correlación directa con la Nota Final ---


not_final             1.000
examen                0.976
nota_cv               0.573
enam                  0.569
total_bonificacion    0.480
Name: not_final, dtype: float64

## Paso 4: Conclusión

A partir del procesamiento del conjunto de datos, se determinó que la muestra analizada está conformada por 17647 registros y 23 variables. La distribución de la condición de ingreso (10564 casos de no ingreso y 7083 casos de ingreso) evidencia un desbalance moderado entre grupos, aspecto metodológicamente relevante para la interpretación comparativa de resultados.

En el análisis descriptivo por condición de ingreso, el grupo que logró ingresar presentó medias superiores en todas las variables académicas evaluadas. En particular, se observaron diferencias favorables en examen (52.233 frente a 47.602), ENAM (1.372 frente a 1.255), nota final (64.790 frente a 59.652), nota curricular (12.557 frente a 12.050) y bonificación total (9.364 frente a 9.016). Asimismo, la varianza de la nota final (76.24) confirma una dispersión apreciable en el rendimiento, lo que sugiere heterogeneidad entre postulantes.

El análisis correlacional de Pearson mostró que la variable examen mantiene la asociación lineal más alta con la nota final (r = 0.976), mientras que la nota curricular (r = 0.573), el ENAM (r = 0.569) y la bonificación total (r = 0.480) exhiben relaciones positivas de magnitud moderada. En consecuencia, los hallazgos respaldan que el desempeño en el examen constituye el principal factor asociado con la variación de la nota final dentro de la cohorte estudiada, sin excluir la contribución complementaria de los demás componentes del proceso de evaluación.